frontend/test/__support__/e2e/helpers/e2e-misc-helpers.js

```diff
@@ -243,3 +243,19 @@ export function saveQuestion(
     cy.button("Not now").click();
   });
 }
+
+export function visitPublicQuestion(id) {
+  cy.request("POST", `/api/card/${id}/public_link`).then(
+    ({ body: { uuid } }) => {
+      cy.visit(`/public/question/${uuid}`);
+    },
+  );
+}
+
+export function visitPublicDashboard(id) {
+  cy.request("POST", `/api/dashboard/${id}/public_link`).then(
+    ({ body: { uuid } }) => {
+      cy.visit(`/public/dashboard/${uuid}`);
+    },
+  );
+}
```

Comment:
Does it make sense to intercept and wait for `GET /api/public/card/:uuid` (and dashboard endpoint too)?

Reply:
Great catch!